asto/asto-s3/src/test/java/com/artipie/asto/s3/S3StorageTest.java

```diff
@@ -100,6 +109,45 @@ void shouldUploadObjectWhenSaveLargeContent(final AmazonS3 client) throws Except
         MatcherAssert.assertThat(this.download(client, key), Matchers.equalTo(data));
     }
 
+    @Test
+    @Timeout(150)
+    void shouldUploadLargeChunkedContent(final AmazonS3 client) throws Exception {
+        final String key = "big/data";
+        final int S3_MIN_PART_SIZE = 5 * 1024 * 1024;
+        final int S3_MIN_MULTIPART = 10 * 1024 * 1024;
+        final int TOTAL_SIZE = S3_MIN_MULTIPART * 2;
+        final int ITEM_SIZE = S3_MIN_PART_SIZE / 1000;
+        final int COUNT = TOTAL_SIZE / ITEM_SIZE;
+        final int TOTAL_SEND = ITEM_SIZE * COUNT;
+        final Random rnd = new Random();
+        final byte[] sentData = new byte[TOTAL_SEND];
+        final Callable<Flowable<ByteBuffer>> getGenerator = () -> Flowable.generate(
+            AtomicInteger::new,
+            (counter, output) -> {
+                final int idx = counter.getAndAdd(1);
+                if (idx < COUNT) {
+                    final byte[] data = new byte[ITEM_SIZE];
+                    rnd.nextBytes(data);
+                    for (int i = 0, j = idx * ITEM_SIZE; i < data.length; ++i, ++j) {
+                        sentData[j] = data[i];
+                    }
+                    output.onNext(ByteBuffer.wrap(data));
+                } else {
+                    output.onComplete();
+                }
+                return counter;
+            });
+        MatcherAssert.assertThat("Generator results mismatch",
+            new PublisherAs(new Content.From(getGenerator.call())).bytes()
```

Comment:
Think, it's enough to check that the size of `sentData` is equal to the expected size.